src/Solid.cpp

```diff
@@ -42,23 +42,15 @@ void CSolid::Options( CDxf * dxf )
 
 void CSolid::Update( cBoundingRectangle& rect )
 {
-    switch (myParser)
-    {
-    case eParser::solid_2point:
-        std::cout << "CSolid::Update " << x <<" "<< y <<" "<< x2 <<" "<< y2 <<"\n";
-        rect.Update( x, y );
-        rect.Update( x2, y2 );
-        break;
-    case eParser::solid_4point:
-        rect.Update( x, y );
-        rect.Update( x1, y1 );
-        rect.Update( x2, y2 );
-        rect.Update( x3, y3 );
-        break;
-    }
+    std::cout << "CSolid::Update " << x <<" "<< y <<" "<< x2 <<" "<< y2 <<"\n";
+    rect.Update( x, y );
+    rect.Update( x2, y2 );
 }
+
 bool CSolid::Append( cvit_t& cvit )
 {
+    double xa[4], ya[4];  // x array, y array, which are used for eParser::solid_4point
```

Comment:
This is EXTREMELY dangerous.

You are relying on the compiler to construct these automatic variables at exaclty the same memory location each time the method is called and to leave them unitialized.  Not all compilers work that way.

Fix: Declare these variables static, so that they are preserved between calls to the method.

Reply:
Hi, James, I think my code is not wrong here.

If you look at the function:

```
bool CSolid::Append( cvit_t& cvit )
{
    double xa[4], ya[4];  // x array, y array, which are used for eParser::solid_4point

    while( true )
    {

    }
```

There is a `while` loop, and the loop exit until we see a `cvit->Code()==0`
Which means the whole solid object is parsed in this `while` loop.

So, no need to static variables, since `CSolid::Append` just read the whole solid element.

Reply:
You are correct.

I forgot how this worked and did not check the code carefully enough,  

I am sorry about that.